internal/cmd/helm-operator/run/cmd.go

```diff
@@ -83,6 +83,7 @@ func NewCmd() *cobra.Command {
 
 	f.AddTo(cmd.Flags())
 	cmd.Flags().AddGoFlagSet(zapfs)
+	cmd.MarkFlagsRequiredTogether("metrics-secure", "metrics-authn-authz")
```

Comment:
This needs to be tweaked. There's a requirement only in one direction. When `metrics-authn-authz` is specified, `metrics-secure` must also be specified. However converse is not true.

There might be existing use of `metrics-secure` in the wild. If we required `metrics-authn-authz` to be specified with it, all of those existing usages would break.

We want to require that the connection is secure if credentials are exchanged. But if credentials are not exchanged, we don't care about whether the connection is secure.

Reply:
So sounds like this would need to be custom `if` logic, since I don't think cobra offers anything like this, unless you know of something?

Reply:
Added the below condition to account for this as an `Args` function for cobra like we do in other cobra cli's for custom flag validation

https://github.com/operator-framework/operator-sdk/pull/6848/commits/c32ccdbf3914685826ef35e7bad00a21d5cf7dbc#diff-d47d8a84d35cfdd54322d503604e6ff0b2f5454d3f1133bfc4faa8faeb9b476dR78-R83